protocols/gossipsub/src/behaviour.rs

```diff
@@ -3169,47 +3174,17 @@ where
         &mut self,
         cx: &mut Context<'_>,
         _: &mut impl PollParameters,
-    ) -> Poll<
-        NetworkBehaviourAction<
-            <Self::ProtocolsHandler as ProtocolsHandler>::InEvent,
-            Self::OutEvent,
-        >,
-    > {
+    ) -> Poll<NetworkBehaviourAction<Self::OutEvent, Self::ProtocolsHandler>> {
         if let Some(event) = self.events.pop_front() {
-            return Poll::Ready(match event {
-                NetworkBehaviourAction::NotifyHandler {
-                    peer_id,
-                    handler,
-                    event: send_event,
-                } => {
-                    // clone send event reference if others references are present
-                    let event = Arc::try_unwrap(send_event).unwrap_or_else(|e| (*e).clone());
-                    NetworkBehaviourAction::NotifyHandler {
-                        peer_id,
-                        event,
-                        handler,
-                    }
-                }
-                NetworkBehaviourAction::GenerateEvent(e) => {
-                    NetworkBehaviourAction::GenerateEvent(e)
-                }
-                NetworkBehaviourAction::DialAddress { address } => {
-                    NetworkBehaviourAction::DialAddress { address }
-                }
-                NetworkBehaviourAction::DialPeer { peer_id, condition } => {
-                    NetworkBehaviourAction::DialPeer { peer_id, condition }
-                }
-                NetworkBehaviourAction::ReportObservedAddr { address, score } => {
-                    NetworkBehaviourAction::ReportObservedAddr { address, score }
-                }
-                NetworkBehaviourAction::CloseConnection {
-                    peer_id,
-                    connection,
-                } => NetworkBehaviourAction::CloseConnection {
-                    peer_id,
-                    connection,
-                },
-            });
+            let event: NetworkBehaviourAction<
+                Self::OutEvent,
+                Self::ProtocolsHandler,
+                Arc<GossipsubHandlerIn>,
+            > = event;
```

Comment:
Based on my local testing, this is unnecessary.
```suggestion
```

Reply:
Oh, a remnant of me debugging the `NetworkBehaviourAction::map_in` logic. As always, much appreciated @thomaseizinger :pray: